sdlf-ddk-lightweight/data_lake/pipelines/common_stages/sdlf_light_transform.py

```diff
@@ -0,0 +1,376 @@
+                            "dynamodb:UpdateItem",
+                            "dynamodb:DeleteItem",
+                            "dynamodb:DescribeTable"
+                        ],
+                        resources=[f"arn:aws:dynamodb:{cdk.Aws.REGION}:{cdk.Aws.ACCOUNT_ID}:table/octagon-*"],
+                    ),
+                    iam.PolicyStatement(
+                        effect=iam.Effect.ALLOW,
+                        actions=[
+                            "kms:CreateGrant",
+                            "kms:Decrypt",
+                            "kms:DescribeKey",
+                            "kms:Encrypt",
+                            "kms:GenerateDataKey*",
+                            "kms:ReEncrypt*",
+                            "kms:ListAliases"
+                        ],
+                        resources=["*"],
+                        conditions={
+                            "ForAnyValue:StringLike": {
+                                "kms:ResourceAliases": [
+                                    f"alias/{self._prefix}-{self.team}-*",
+                                    f"alias/{self._prefix}-octagon-*"
+                                ]
+                            }
+                        }
+                    ),
+                    iam.PolicyStatement(
+                        effect=iam.Effect.ALLOW,
+                        actions=[
+                            "ssm:GetParameter",
+                            "ssm:GetParameters"
+                        ],
+                        resources=[f"arn:aws:ssm:{cdk.Aws.REGION}:{cdk.Aws.ACCOUNT_ID}:parameter/*"],
+                    ),
+                    iam.PolicyStatement(
+                        effect=iam.Effect.ALLOW,
+                        actions=[
+                            "s3:Get*",
+                            "s3:List*",
+                            "s3-object-lambda:Get*",
+                            "s3-object-lambda:List*"
+                        ],
+                        resources=["*"],
+                    ),
+                    iam.PolicyStatement(
+                        effect=iam.Effect.ALLOW,
+                        actions=[
+                            "sqs:SendMessage",
+                            "sqs:GetQueueAttributes",
+                            "sqs:ListQueues",
+                            "sqs:GetQueueUrl",
+                            "sqs:ListDeadLetterSourceQueues",
+                            "sqs:ListQueueTags"
+                        ],
+                        resources=[f"arn:aws:sqs:{cdk.Aws.REGION}:{cdk.Aws.ACCOUNT_ID}:{self._prefix}-{self.team}-*"],
+                    ),
+                    iam.PolicyStatement(
+                        effect=iam.Effect.ALLOW,
+                        actions=[
+                            "states:StartExecution"
+                        ],
+                        resources=[f"arn:aws:states:{cdk.Aws.REGION}:{cdk.Aws.ACCOUNT_ID}:stateMachine:*"],
```

Comment:
Minor nitpick: do we want this routing lambda to only be able to start any state machine or just the state machine for this light transform?

Reply:
fixed

Reply:
original fix (`self._state_machine.grant_start_execution(self._lambda_role)`) created a circular dependency so I went back to a hardcoded policy but updated the policy to only allow start execution on sdlf state machines. The DDK `StateMachineStage` doesn't allow setting the state machine name explicitly so I think this is a good compromise.